develop-docs/self-hosted/troubleshooting/kafka.mdx

```diff
@@ -4,19 +4,19 @@ sidebar_title: Kafka
 sidebar_order: 2
 ---
 
-## Offset Out Of Range Error
+## How Kafka Works
 
-```log
-Exception: KafkaError{code=OFFSET_OUT_OF_RANGE,val=1,str="Broker: Offset out of range"}
-```
+This section is aimed for those who have Kafka problems, but are not yet familiar with Kafka. At a high level, it is a message broker which stores message in a log (or in an easier language: very similar to an array) format. It receives messages from producers that aimed to a specific topic, and then sends them to consumers that are subscribed to that topic. The consumers can then process the messages.
 
-This happens where Kafka and the consumers get out of sync. Possible reasons are:
+On the inside, when a message enters a topic, it would be written to a certain partition. You can think partition as physical boxes that stores messages for a specific topic, each topic will have their own separate & dedicated partitions. In a distributed Kafka setup, each partition might be stored on a different machine/node, but if you only have a single Kafka instance, then all the partitions are stored on the same machine.
```

Comment:
This paragraph has a bit of overlap with the next one and does not add much to understanding kafka IMO, so I think we could remove it completely.